lib/Facebook.js

```diff
@@ -1876,6 +1876,21 @@ function Facebookbot(configuration) {
         }
     }
 
+    function scopeAPIsWithRuntimeConfiguration(configuration) {
```

Comment:
i put this here as one possible idea for how to allow for passing a spawn config to the api methods to allow for runtime page-access-token configuration.  i'm totally open to other ways of solving this, it was just my first thought for having to change the least amt of code

Reply:
looking at this a bit more, it looks like this could easily work for associating a config with all the api methods, however, tracing the path of where we have the `.defineBot` happening, i believe it's the global `botkit` facebook object getting passed in as the first param of the method passed in to `.defineBot`. Because of this, reassigning `bot.botkit.api = scopAPIsWithRuntimConfiguration(bot.config)` would have an unintended side affect of modifying it across a shared reference. 
not sure if it's worth assigning `bot.scopedApi = scopeAPIsWithRuntimeConfiguration(bot.config)`. while it would add a second api field on the spawned bot worker, it would also prevent any weirdness with mutating a global reference.  I don't have any other ideas at the moment, but will keep thinking on it. lmk what you think 😄